datafusion/core/src/physical_plan/planner.rs

```diff
@@ -929,10 +972,11 @@ impl DefaultPhysicalPlanner {
                     }
 
                     // All equi-join keys are columns now, create physical join plan
+                    let mut left_right = self.create_initial_plan_multi([left.as_ref(), right.as_ref()], session_state).await?;
```

Comment:
I'd love to see this not `unwrap` - perhaps match on a slice?

Reply:
> what do you do then?

I would recommend:

```
Err(DataFusionError::PlanError("xyz")?
```

Reply:
I don't agree w/ on this one. Fundamentally broken stuff can panic, that's a common pattern in Rust. Errors should only be used where it makes sense. Esp. most errors don't even carry enough information for a user to be meaningful.

Anyways, I'm not gonna die on that hill. Fixed.